spec/services/user_signup_spec.rb

```diff
@@ -0,0 +1,138 @@
+require 'spec_helper'
+
+describe UserSignup do
+
+  describe '#sign_up' do
+    let(:stripe_charge) { double(Stripe::Charge) }
+    let(:service) { UserSignup.new(user) }
+
+    before do
+      allow(StripeWrapper::Charge).to receive(:create).and_return stripe_charge
+      allow(stripe_charge).to receive_messages(succesful?: true, customer_id: "ABC123")
+    end
+
+    context 'with valid input' do
+      let(:user) { Fabricate.build(:user) }
+
+      it 'sets the status to success' do
+        service.sign_up
```

Comment:
It is true that in this spec it is not needed, because of that stub. But specs are supposed to be description or 'specialization' of your implementation code. When you omit the token, it actually causes me to understand that the token is not required in first place. It is better to show the intention, and here the service won't work without token being passed in implementation.